doc/src/gotchas.rst

```diff
@@ -758,6 +758,103 @@ that you want, and they will all be evaluated according to the function.
 
     See the Python docs for more information on function parameters.
 
+
+Classification of the objects
+=============================
+
+There are several ways of how SymPy object is classified.
+
+class
+-----
+
+Like any other object in Python, SymPy expression is an instance of class. You can
+get the class of the object with built-in `type()` function, and check it with
+`isinstance()` function.
+
+    >>> type(x + y)
+    <class 'sympy.core.add.Add'>
+    >>> isinstance(x + y, Add)
+    True
+
+Classes represent only the programmatic structures of the objects, and does not
+distinguish the mathematical difference between them. For example, the integral
+of number and the integral of matrix both have the class `Integral`, although the
+former is number and the latter is matrix.
+
+    >>> A = MatrixSymbol('A', 2, 2)
+    >>> type(Integral(1, x))
+    <class 'sympy.integrals.integrals.Integral'>
+    >>> type(Integral(A, x))
+    <class 'sympy.integrals.integrals.Integral'>
+
+kind
+----
+
+Kind indicates what mathematical object does the expression represent.
+You can retrieve the kind of expression with `sympy.core.kind()` function.
+
+    >>> from sympy.core import kind
```

Comment:
If we have a kind function then it should be at top-level import like `from sympy import kind`